src/Settings.ts

```diff
@@ -32,6 +33,12 @@ export namespace Settings {
         workspace.getConfiguration().update("maven.view", "hierarchical", false);
     }
 
+    export function storeFavorite(favorite: FavoriteFormat): void {
+        const favorites: FavoriteFormat[] | undefined = workspace.getConfiguration().get("maven.terminal.favorites");

```

Comment:
although default value of `maven.terminal.favorites` is `[]` (defined in package.json), here according to vscode's api, its value is possible to be `undefined`. 

Suggest to change as below
```
const favorites: FavoriteFormat[] = workspace.getConfiguration().get("maven.terminal.favorites") ?? []; 
favorites.push(favorite);
```